packages/amesos2/src/Amesos2_KLU2_def.hpp

```diff
@@ -252,6 +252,10 @@ KLU2<Matrix,Vector>::numericFactorization_impl()
            data_.symbolic_, &(data_.common_)) ;
       } //end single_process_optim_check = false
 
+      if(data_.numeric_ == NULL) {
+          throw std::runtime_error("KLU2 numeric factorization failed.");
```

Comment:
`TEUCHOS_TEST_FOR_EXCEPTION` can give better information, like caller info, a hook for setting a debugger break point, and an optional stack trace.
```c++
TEUCHOS_TEST_FOR_EXCEPTION
  (data_.numeric_ == nullptr, std::runtime_error,
   "KLU2 numeric factorization failed");
```

Reply:
Updated. Thanks.